### ISSUE TITLE
flow diagram

### ISSUE THREAD

**Issue by nschneid** [nschneid/activedop#93]
Super fancy attempt at a diagram:

![image](https://github.com/user-attachments/assets/f3c57c4f-36ee-466e-8579-4f2c8e4ffb50)

This is based on my rough understanding of the code. Is it a good and accurate summary of the stages in the tool workflow and how it interacts with data?

E.g., it claims that

- the database is written to only in initdb, accept tree, and delete tree operations
- the database is read only by `annotate()`
- the parser is called only by initpriorities and reparse operations (including reparse-subtree)

**Comment by bwaldon** [comment]
This looks mostly accurate to me! A couple notes: 

- Since https://github.com/nschneid/activedop/commit/b4b331812f7ae1ee7a191476b997659cafb5d8d7, the database is read by `initpriorities` (s/t sentences with saved annotations are not re-parsed). 

- In addition to being called by `initpriorities` and `reparse`, the parser is also called (re-trained) when a user accepts a tree: https://github.com/bwaldon/activedop/blob/d3c8dc486051e96e8469e8c9793226c34fce2565/app.py#L1182

- Not part of the main workflow, but the `get_data_psv` endpoint also reads the database to export all annotations.

**Comment by nschneid** [comment]
Updated
![image](https://github.com/user-attachments/assets/a8e15ffd-f04c-47d2-b8b3-0cfb2da4c9b2)

**Comment by nschneid** [comment]
With #94 I think there's technically another database read from edit view (upon loading a saved tree).

**Comment by bwaldon** [comment]
That's right. And when there is no saved tree`edit()` calls the parser: https://github.com/nschneid/activedop/blob/fc4f39bf239d1bbe4f224ab37d79616df395eecf/app.py#L683
